src/libmongoc/tests/test-libmongoc.h

```diff
@@ -98,10 +98,18 @@ void
 test_framework_set_ssl_opts (mongoc_client_t *client);
 void
 test_framework_set_pool_ssl_opts (mongoc_client_pool_t *pool);
+
+mongoc_client_t *
+test_framework_new_default_client (void);
+mongoc_client_t *
+test_framework_client_new (const char *uri_str);
 mongoc_client_t *
-test_framework_client_new (void);
+test_framework_client_new_from_uri (const mongoc_uri_t *uri);
+
+mongoc_client_pool_t *
+test_framework_new_default_client_pool (void);
 mongoc_client_pool_t *
-test_framework_client_pool_new (void);
+test_framework_client_pool_new (const mongoc_uri_t *uri);
```

Comment:
I dislike that this does not match the signature of `test_framework_client_new` but rather `test_framework_client_new_from_uri`.

My suggestion would be:
```suggestion
test_framework_client_pool_new (const char *uri_str);
mongoc_client_t *
test_framework_client_pool_new_from_uri (const mongoc_uri_t *uri);
```

Reply:
The reason I changed this to take a uri was to match the signature of `mongoc_client_pool_new`, which takes a uri:

```
MONGOC_EXPORT (mongoc_client_pool_t *)
mongoc_client_pool_new (const mongoc_uri_t *uri);
```

I find it confusing that the test framework doesn't match the real class' methods.  It's also a bit weird that `mongoc_client_pool_t` and `mongoc_client_t` have methods whose signatures don't agree, but that's a discussion for another day.

Do you think it's more important that `test_framework_client_pool_*` agrees with `mongoc_client_pool_*`, or that it agrees with `test_framework_client_*` ? Unfortunately we can't have both :(

Reply:
I think it's more natural to have the `test_framework_client_*` methods have consistent signatures and deal with any API differences internally. This at least fixes the wrong signature issue in tests, leaving us to figure out the actual API on a different day.